test/unit/preprocess.lua

```diff
@@ -145,21 +149,40 @@ end
 
 -- returns a stream representing a preprocessed form of the passed-in headers.
 -- Don't forget to close the stream by calling the close() method on it.
-function Gcc:preprocess_stream(...)
+function Gcc:preprocess(...)
   -- create pseudo-header
   local pseudoheader = headerize({...}, false)
+  local pseudoheader_fname = 'tmp_pseudoheader.h'
+  local pseudoheader_file = io.open(pseudoheader_fname, 'w')
+  pseudoheader_file:write(self.added_header_defines)
+  pseudoheader_file:write("\n")
+  pseudoheader_file:write(pseudoheader)
+  pseudoheader_file:flush()
+  pseudoheader_file:close()
   local defines = table.concat(self.preprocessor_extra_flags, ' ')
   local cmd = ("echo $hdr | " ..
                tostring(self.path) ..
                " " ..
                tostring(defines) ..
-               " -std=c99 -P -E -"):gsub('$hdr', shell_quote(pseudoheader))
+               " -std=c99 -P -E " .. shell_quote(pseudoheader_fname))
+  local def_cmd = ("echo $hdr | " ..
```

Comment:
`echo $hdr | … -` was replaced with actual file name. Just forgot to remove the command.